src/main/java/net/snowflake/client/jdbc/SnowflakeColumnMetadata.java

```diff
@@ -20,12 +21,45 @@ public class SnowflakeColumnMetadata implements Serializable {
   private int scale;
   private boolean fixed;
   private SnowflakeType base;
+  private List<FieldMetadata> fields;
   private String columnSrcTable;
   private String columnSrcSchema;
   private String columnSrcDatabase;
 
   private boolean isAutoIncrement;
 
+  public SnowflakeColumnMetadata(
+      String name,
+      int type,
+      boolean nullable,
+      int length,
+      int precision,
+      int scale,
+      String typeName,
+      boolean fixed,
+      SnowflakeType base,
+      List<FieldMetadata> fields,
+      String columnSrcDatabase,
+      String columnSrcSchema,
+      String columnSrcTable,
+      boolean isAutoIncrement) {
+    this.name = name;
+    this.type = type;
+    this.nullable = nullable;
+    this.length = length;
+    this.precision = precision;
+    this.scale = scale;
+    this.typeName = typeName;
+    this.fixed = fixed;
+    this.base = base;
+    this.fields = fields;
+    this.columnSrcDatabase = columnSrcDatabase;
+    this.columnSrcSchema = columnSrcSchema;
+    this.columnSrcTable = columnSrcTable;
+    this.isAutoIncrement = isAutoIncrement;
+  }
+
+  @Deprecated
```

Comment:
let's add javadoc with note which constructor should be used instead

Reply:
ok